Forwarded: not-needed
Origin: https://patch-diff.githubusercontent.com/raw/MariaDB/server/pull/2777.patch
Bug: https://github.com/MariaDB/server/pull/2777
From: Daniel Black <daniel@mariadb.org>
Date: Thu, 5 Oct 2023 16:55:34 +0300
Subject: [PATCH] MDEV-21587: disk.disk{_notembedded} test result

Allow for a CI system to be almost out of space, or having so
little use, that the Total space is the same as available or used.

Thanks Otto Kekäläinen for the bug report and testing.
---
 plugin/disks/mysql-test/disks/disks.result             | 4 ++--
 plugin/disks/mysql-test/disks/disks.test               | 2 +-
 plugin/disks/mysql-test/disks/disks_notembedded.result | 8 ++++----
 plugin/disks/mysql-test/disks/disks_notembedded.test   | 4 ++--
 4 files changed, 9 insertions(+), 9 deletions(-)

--- a/plugin/disks/mysql-test/disks/disks.result
+++ b/plugin/disks/mysql-test/disks/disks.result
@@ -7,6 +7,6 @@ DISKS	CREATE TEMPORARY TABLE `DISKS` (
   `Used` bigint(32) NOT NULL,
   `Available` bigint(32) NOT NULL
 ) ENGINE=MEMORY DEFAULT CHARSET=utf8mb3 COLLATE=utf8mb3_general_ci
-select sum(Total) > sum(Available), sum(Total)>sum(Used) from information_schema.disks;
-sum(Total) > sum(Available)	sum(Total)>sum(Used)
+select sum(Total) >= sum(Available), sum(Total)>=sum(Used) from information_schema.disks;
+sum(Total) >= sum(Available)	sum(Total)>=sum(Used)
 1	1
--- a/plugin/disks/mysql-test/disks/disks.test
+++ b/plugin/disks/mysql-test/disks/disks.test
@@ -1,3 +1,3 @@
 --replace_regex /varchar\([0-9]+\)/varchar(pathlen)/
 show create table information_schema.disks;
-select sum(Total) > sum(Available), sum(Total)>sum(Used) from information_schema.disks;
+select sum(Total) >= sum(Available), sum(Total)>=sum(Used) from information_schema.disks;
--- a/plugin/disks/mysql-test/disks/disks_notembedded.result
+++ b/plugin/disks/mysql-test/disks/disks_notembedded.result
@@ -6,16 +6,16 @@ CREATE USER user1@localhost;
 GRANT SELECT ON *.* TO user1@localhost;
 connect  con1,localhost,user1,,;
 connection con1;
-select sum(Total) > sum(Available), sum(Total)>sum(Used) from information_schema.disks;
-sum(Total) > sum(Available)	sum(Total)>sum(Used)
+select sum(Total) >= sum(Available), sum(Total) >= sum(Used) from information_schema.disks;
+sum(Total) >= sum(Available)	sum(Total) >= sum(Used)
 NULL	NULL
 disconnect con1;
 connection default;
 GRANT FILE ON *.* TO user1@localhost;
 connect  con1,localhost,user1,,;
 connection con1;
-select sum(Total) > sum(Available), sum(Total)>sum(Used) from information_schema.disks;
-sum(Total) > sum(Available)	sum(Total)>sum(Used)
+select sum(Total) >= sum(Available), sum(Total) >= sum(Used) from information_schema.disks;
+sum(Total) >= sum(Available)	sum(Total) >= sum(Used)
 1	1
 connection default;
 DROP USER user1@localhost;
--- a/plugin/disks/mysql-test/disks/disks_notembedded.test
+++ b/plugin/disks/mysql-test/disks/disks_notembedded.test
@@ -10,7 +10,7 @@ GRANT SELECT ON *.* TO user1@localhost;

 connect (con1,localhost,user1,,);
 connection con1;
-select sum(Total) > sum(Available), sum(Total)>sum(Used) from information_schema.disks;
+select sum(Total) >= sum(Available), sum(Total) >= sum(Used) from information_schema.disks;
 disconnect con1;

 connection default;
@@ -18,7 +18,7 @@ GRANT FILE ON *.* TO user1@localhost;

 connect (con1,localhost,user1,,);
 connection con1;
-select sum(Total) > sum(Available), sum(Total)>sum(Used) from information_schema.disks;
+select sum(Total) >= sum(Available), sum(Total) >= sum(Used) from information_schema.disks;
 connection default;
 DROP USER user1@localhost;
